### DISCUSSION TITLE
Question regarding expected behavior of TE.fromTask

### DISCUSSION THREAD

**Discussion by niklaserik** [gcanti/fp-ts#1791] [General]
Hi,

I want to start to thank you for a great library!

My question is regarding the type of the natural transformation `TaskEither`'s `fromTask` type. I have a chain of TaskEithers and hence I want to convert a my Task into a TaskEither to get a nice flow in the code. Since a Task never is supposed to fail, I expected the type after the transformation to be `TE.TaskEither<never, T>`. However, the transformation resulted in the type `TE.TaskEither<unknown, T>`

I can achieve the expected behavior by making an anonymous function like `TE.chainW(flow(functionReturningTask, (a) => TE.fromTask<T, never>(a)))` but it would be nice to just be able to write `TE.chainW(flow(functionReturningTask, TE.fromTask ))`

Perhaps there is a reason I an not aware about why it converts it to `unknown´ instead of `never`?

**Comment by gcanti** [comment]
@niklaserik that's how it should work, could you provide a repro?

**Comment by niklaserik** [comment]
@gcanti check the types of the anonymous functions in the dummy code below;

```
import { flow, pipe } from 'fp-ts/lib/function';
import * as T from 'fp-ts/Task';
import * as E from 'fp-ts/Either';
import * as TE from 'fp-ts/TaskEither';
import axios from 'axios';


const fetch = TE.tryCatch(()=>axios.get('https://google.com'),E.toError);
const task = (a: unknown):T.Task<void>  => ()=> Promise.resolve();
const a = pipe(
    fetch, 
    b=>b, 
    TE.chainW(flow(task, c=>c, TE.fromTask, d=>d )),
    f=>f,
);
```
Getting the same issue with `unknown` instead of `never` there.
I'm running it with `"fp-ts": "2.12.2"` and `"typescript": "4.6.4"`

**Reply by gcanti** [reply]
Looks like the culprit is using `fromTask` with `flow`.
Should be fixed in `fp-ts@2.13` (currently released as an RC)

However regardless of the version you are using, my advice would be to use the `chainTaskK` helper (basically `chain` + `fromTask` in one shot)

```ts
// const a: TE.TaskEither<Error, void>
const a = pipe(fetch, TE.chainTaskK(task))
```

**Reply by JoshuaAmaju** [reply]
How is `chainTaskK` different from `chainTask`. I've seen a lot of that variation for Option etc, what's the difference?

**Comment by niklaserik** [comment]
Thank you for the advice and the awesome library. Works perfectly.
